devito/function.py

```diff
@@ -1089,3 +1145,25 @@ def __indices_setup__(cls, **kwargs):
     @classmethod
     def __shape_setup__(cls, **kwargs):
         return kwargs.get('shape', (kwargs.get('nt'), kwargs.get('npoint'),))
+
+    @property
+    def dt(self):
```

Comment:
Glad my rewrite of the hierarchy a few weeks ago is helping.

As pointed out in #484 , I think these methods should be dynamically (and conditionally) built within ``_initialize_derivatives``

Reply:
It would need `_initialize_derivatives` to know about `TimeSparseFunction ` in `SparseFunction` which is a bit odd no?